lib/events-reader.js

```diff
@@ -74,6 +80,22 @@ module.exports = function(harvesterApp) {
       var doc;
       doc = that.stream.read();
 
+      // If it is a doc add to kafka
+      if(doc !== null && (!doc.ns.includes('checkpoint'))) {
+        producer.init().then(function(){
+          return producer.send({
+            topic: 'darksouls',
+            partition: 0,
+            message: {
+              value: JSON.stringify(doc)
+            }
+          });
+        })
+        .then(function (result) {
+          console.log('>>> producer sent message');
+        });
+      }
+
       var promises = that.processDocHandlers(doc);
```

Comment:
From my understanding, correct me if I'm wrong, the `processDocHandlers` function calls those functions defined on the `onChange` hook from each resource and return promises which will be resolved when the those functions were actually triggered.

If this is true, actually the `processDocHandlers` function it would be on the `events-consumer` file right after we consume some message from kafka. In that case, in fact `events-reader` would only tail the oplog and stream it to kafka when is necessary.

Is that correct?